### PR TITLE
carbon component with China PIPL changes

### PR DESCRIPTION
**Pull request by deathcave**
### Related Ticket(s)

[Noitce & Choice w3 Publisher](https://w3.ibm.com/w3publisher/urx/notice-and-choice-v23)

### Description

Notice & Choice is a legally required component to create a form where IBM collects its customer information.
This section shows the specific product's legal links, terms, and conditions. Along with it, it collects users' consent regarding communication preferences.

### Changelog

**New**

-  A new web component to create notice & choice user interface.

**Changed**

-  Added China PIPL 

**Removed**

- Postal option removed from new notice and choice version

### CODE REVIEW

**Comment by deathcave** [comment]
I have read the DCO document and I hereby sign the DCO.

**Review by sanjitbauli — APPROVED** [review]

**Review by sanjitbauli — CHANGES_REQUESTED** [review]
@deathcave please add a flag to turn on/off the country-based legal checkbox. this will help control the behaviour from the adopter's page

**Comment by sanjitbauli** [comment]
@deathcave also add a part to the paragraph under ncPostTextContainer (terms and condition sentence)

File: packages/web-components/src/components/notice-choice/notice-choice.ts [hunk 1]
```diff
@@ -299,6 +316,69 @@ class NoticeChoice extends StableSelectorMixin(LitElement) {
     return `${ddsPrefix}--notice-choice`;
   }
   static styles = styles; // `styles` here is a `CSSResult` generated by custom WebPack loader
+  checkBoxLegalChange($event: any) {
+    const legalCheckbox = $event.target;
+    const isChecked = legalCheckbox.checked ? true : false;
+    const legalTextError = legalCheckbox.parentNode.querySelector('.nc-error');
+    legalCheckbox.value = isChecked ? 1 : 0;
+    legalTextError.style.display = isChecked ? 'none' : '';
```
Comment (sanjitbauli) [inline packages/web-components/src/components/notice-choice/notice-choice.ts:324]: @deathcave if hide-error-message is true, in that case, legalTextError is null, you cannot access style.display.
add an additional check here.

**Review by sanjitbauli — CHANGES_REQUESTED** [review]

File: packages/web-components/src/components/notice-choice/notice-choice.ts (hunk 1 above)
Comment (deathcave) [reply packages/web-components/src/components/notice-choice/notice-choice.ts]: Fixed Please check

**Comment by deathcave** [comment]
> @deathcave also add a part to the paragraph under ncPostTextContainer (terms and condition sentence)

Added

**Comment by kennylam** [comment]
The CI is failing the formatting checks. you can do a `yarn format` from project root to fix.

**Comment by deathcave** [comment]
> The CI is failing the formatting checks. you can do a `yarn format` from project root to fix.

Fixed

**Review by sanjitbauli — APPROVED** [review]

**Review by kennylam — APPROVED** [review]

**Pull request merged**